endpointmanager/pkg/chplendpointquerier/onemedicalwebscraper.go

```diff
@@ -0,0 +1,80 @@
+package chplendpointquerier
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/PuerkitoBio/goquery"
+	log "github.com/sirupsen/logrus"
+	"io/ioutil"
+	"strings"
+
+	"github.com/chromedp/chromedp"
+)
+
+func oneMedicalWebscraper(oneMedicalURL string, fileToWriteTo string) {
+
+	var lanternEntryList []LanternEntry
+	var endpointEntryList EndpointList
+
+	ctx, cancel := chromedp.NewContext(context.Background())
+	defer cancel()
+
+	var htmlContent string
+
+	// Chromedp will wait for webpage to run javascript code to generate api search results before grapping HTML
+	err := chromedp.Run(ctx,
+		chromedp.Navigate(oneMedicalURL),
+		chromedp.WaitVisible("root-url"),
+		chromedp.OuterHTML("html", &htmlContent, chromedp.ByQuery),
+	)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	doc, err := goquery.NewDocumentFromReader(strings.NewReader(htmlContent))
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	doc.Find("main").Each(func(index int, mainElem *goquery.Selection) {
```

Comment:
A lot of the web scraper functions seem to be the same, with this piece where it actually finds the url(s) in the webpage being the main difference. Do you think it's worth trying to pull out everything up to this line and everything at line 68 and below to it's own separate function?